[#4][news] feat: 뉴스 init

luz315 opened the pull request:
## Description
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

Resolves: #(Isuue Number)

## PR Type
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

luz315 committed:
[#4][news] feat: 뉴스 init

luz315 committed:
Merge branch 'develop' into feature/news/init

luz315 committed:
[#4][news] feat: 뉴스 구조 컨벤션 통일

luz315 committed:
[#4][news] feat: 뉴스관련 엔티티 작성

venus-y commented on news/src/main/java/com/newpick4u/news/news/domain/entity/News.java:
```diff
@@ -0,0 +1,44 @@
+@Table(name = "p_news")
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
```
JPA에서 엔티티의 기본 생성자를 요구하고 접근 제어자는 public 또는 protected 범위 내에 있어야 하기 때문에 접근 범위를 최대한 좁히면서 JPA를 사용하는 데 지장이 없도록 protected로 설정하는 것이 좋아보입니다. 전체 필드 생성자는 JPA에서 필수적이지 않으므로, 이를 private으로 설정하여 접근 범위를 최대한 좁히는 것이 좋을 것 같습니다.

luz315 replied:
수정해두었습니다~

hp5234 commented on news/src/main/java/com/newpick4u/news/news/domain/entity/News.java:
```diff
@@ -0,0 +1,44 @@
+    @BatchSize(size = 10)
+    private List<NewsTag> tagList = new ArrayList<>();
+
+    public static News create(String title, String content, Long view) {
```
메서드 명을 from, of 처럼 사용하면 조금 더 명확하게 의미전달이 될것같습니다. 
![image](https://github.com/user-attachments/assets/871df6ed-ef0d-493d-9d37-ee456913003d)

luz315 replied:
값 조합으로 객체 생성하거나 다른 객체로부터 변환하는 정적 팩토리 메서드를 주로 사용하는 디티오 단에서는 from, of를 많이 사용하지만 도메인 계층에서는 새로운 객체를 "직접 정의해서" 생성하는 의미에 초점을 맞추면 create도 많이 사용된다고 들었습니다 
오히려 from 이나 of 보다는 create의 개념이 더 맞지않나 생각하는데 해당 주제로 회의해도 좋을 것 같습니다

hp5234 commented on news/src/main/java/com/newpick4u/news/news/domain/entity/NewsTag.java:
```diff
@@ -0,0 +1,33 @@
+    @JoinColumn(name = "news_id", nullable = false, unique = true)
+    private News news;
+
+    public static NewsTag create(String name, News news) {
```
메서드 명을 from, of 처럼 사용하면 조금 더 명확하게 의미전달이 될것같습니다. 
![image](https://github.com/user-attachments/assets/871df6ed-ef0d-493d-9d37-ee456913003d)

luz315 replied:
값 조합으로 객체 생성하거나 다른 객체로부터 변환하는 정적 팩토리 메서드를 주로 사용하는 디티오 단에서는 from, of를 많이 사용하지만 도메인 계층에서는 새로운 객체를 "직접 정의해서" 생성하는 의미에 초점을 맞추면 create도 많이 사용된다고 들었습니다 
오히려 from 이나 of 보다는 create의 개념이 더 맞지않나 생각하는데 해당 주제로 회의해도 좋을 것 같습니다

luz315 committed:
[#4][news] refactor: yml 셋팅 .env 추가

luz315 committed:
[#4][news] refactor: 엔티티 UUID 컨벤션 통일, PROTECTED 설정

luz315 committed:
[#4][news] feat: ai-news 카프카 설정

- 카프카 컨슈머 구현
- Consumer 옵셋 수동커밋 처리
- DLQ 기반 장애복구 로직 구현
- retry 정책, error handler, poll timeout, heartbeat, 역직렬화 설정

luz315 committed:
[#4][news] feat: 리포지토리 dip 준수

luz315 committed:
[#4][news] feat: 태그 클라이언트 구현

luz315 committed:
[#4][news] refactor: retry 로직 자동 역직렬화 추가

luz315 committed:
[#4][news] feat: 카프카 통신 후 뉴스 자동 업로드 로직 구현

luz315 committed:
[#4][news] refactor: 불필요 로직 제거

luz315 committed:
[#4][news] feat: NewsMessageHandler 추가

luz315 committed:
[#4][news] refactor: 카프카 설정 값 로직 분리

luz315 committed:
[#4][news] refactor: 내부 빌더 생성하여 정적메서드 정의

venus-y commented on news/src/main/java/com/newpick4u/news/news/infrastructure/feign/TagClient.java:
```diff
@@ -0,0 +1,15 @@
+public interface TagClient {
+
+    @GetMapping ("/api/v1/tag/list")
+    List<TagDto> getOrCreateTags(@RequestBody List<String> tagNames);
```
`@GetMapping ("/api/v1/tag/list")` -> `@GetMapping ("/api/v1/tags/list")`가 되어야 할 것 같스빈다.

luz315 replied:
해당부분은 예시로 작성된건데 말씀하신대로 tags로 바꿔놓았습니다
다만 페인클라이언트를 할 지도, 한다면 태그 측에서 api url이 어떻게 생성될 지 몰라 일단 참고로만 봐주시면 좋을 것 같습니다 :)

luz315 committed:
[#4][news] refactor: 클라이언트 url 변경

venus-y reviewed (approved):
고생하셨습니다.
